### ISSUE TITLE
Implement Reference.contextualize

### ISSUE THREAD

**Issue by hongsudt** [informatics-isi-edu/ermrestjs#91]

**Comment by RFSH** [comment]
Most of the variables that are under `Reference` would be affected by `.contextualize`. 

One approach is changing everything based on context in this function. This approach requires unnecessary preprocessing.

As an alternative, we can define a `_context` in `Reference` and just change `_context` in `.contextualized`. This way, each attribute that is context-dependent, is responsible for using `_context` to return the *contextualized* value.

```javascript
  Reference.prototype = {
    ...
  
    contexualize: {
      get record (){
        var newRef = module._referenceCopy(this); // it should not copy the context related stuffs?

         //context that is defined here should be the same as it is defined in annotation
        newRef._context = "record";

        return newRef;
      },
      ...
    },

    get columns(){
      if(this._columns !== undefined){
        return this._columns;
      }
      return this._table.columns._contextualize(this._context);
    },
    
    ...
  }  
```

**Comment by jrchudy** [comment]
This issue is resolved by PR #115.
